src/core/p5.Renderer2D.js

```diff
@@ -17,6 +17,19 @@ class Renderer2D extends Renderer {
     super(elt, pInst, isMainCanvas);
     this.drawingContext = this.canvas.getContext('2d');
     this._pInst._setProperty('drawingContext', this.drawingContext);
+    this.elt = elt;
+
+    // Extend renderer with methods of p5.Element with getters
+    this.wrappedElt = new p5.Element(elt, pInst);
+    for (const p of Object.getOwnPropertyNames(p5.Element.prototype)) {
+      if (p !== 'constructor' && p[0] !== '_') {
+        Object.defineProperty(this, p, {
+          get() {
+            return this.wrappedElt[p];
+          }
+        })
+      }
+    }
```

Comment:
For changing the hierachy of the OOP structure, `p5.Element` is created as a property `wrappedElt` of `p5.Renderer2D` as `p5.Renderer` may not need any DOM nodes. This creates a problem which is that p5.Element methods are expected to be present in the renderer object (returned by `createCanvas()` for example) so that one may call `.id()` or `.mousePressed()` on it.

This change above more or less preserve the existing behavior outwardly by aliasing `p5.Renderer.id()` to `p5.Renderer.wrappedElt.id()` using getter (may not be necessary and can just be directly assigned). This gives `p5.Renderer2D` the appearance of being `p5.Element` but not technically an instance of it.

Another less magical option is to make a breaking change so that the renderer's underlying `p5.Element` can only be accessed through a property of the renderer, and so getting the id of the element backing a renderer would be more like `canvas.element.id()` instead of the current `canvas.id()`. This option has potential to break quite a bit of existing sketches though as it is quite common to do `canvas.parent("#some-div")` to place a canvas in the DOM.

Reply:
I'm ok with either of those (maybe we could make a `mixin()` function so that other renderers can easily do the same thing if we go with the first option?)

Another thought: do users expect a renderer to come out of `createCanvas` or could that return the element of the renderer? Off the top of my head I can't think of (public) APIs on a renderer itself that aren't just on `p5.Element` that people might be using. I suppose not all renderers have an element to return though (maybe `undefined` is a valid return type here?)

Reply:
Hmm...perhaps returning p5.Element for those using it could be something `createCanvas` returns as it aligns with the other `createX()` functions in the DOM module as well. However in this case I guess the created renderer will not be available publicly necessarily then.

Reply:
Right, I think that's OK if renderers are private if renderers have a way of adding public methods to p5. So things like sphere() could exist just in the WebGL renderer, added to p5

Reply:
Yeah, that one is somewhere on my list for this PR, might go with the existing method (defining `p5.prototype.sphere()` that calls `this._renderer.sphere()` or might go with the getter technique here as well. We can discuss when I get to that implementation.